lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -1247,6 +1247,14 @@ static void populateTypeConversion(TypeConverter &typeConverter) {
                               type.getSize());
   });
 
+  // FIXME: In order to avoid the stack error caused by the conversion of
+  // "RefType" which can return "{}", such as "ref<uarray<4xi4>>", we can
+  // convert "moore::uarray" into "hw::array" temporarily.
+  typeConverter.addConversion([&](UnpackedArrayType type) {
+    return hw::UnpackedArrayType::get(
+        typeConverter.convertType(type.getElementType()), type.getSize());
```

Comment:
We should check if the type returned by `convertType` is NULL and in that case forward the NULL rather than calling `UnpackedArrayType::get` with it (which could cause another segfault).

Reply:
I quite agree with you. I checked the location of the segfault, and all point to `moore.variable`. I think I can close this PR, because no operation can handle `hw::UnpackedArrayType` now.